lib/theme_check/html_node.rb

```diff
@@ -22,35 +22,57 @@ def element?
     end
 
     def children
-      @value.children.map { |child| HtmlNode.new(child, template) }
+      @value.children.map { |child| HtmlNode.new(child, template, @placeholder_values) }
```

Comment:
I would memoize all these to avoid replacing each time the method is called.

```suggestion
      @children ||= @value.children.map { |child| HtmlNode.new(child, template, @placeholder_values) }
```

Same w/ `attributes`, `parent`, etc. bellow.